File: 078/cpp/main.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        if (nums.empty()) {
            return {{}};
        }
        int n = nums.back();
        nums.pop_back();
        auto result = subsets(nums);
        int size = result.size();
        for (int i = 0; i < size; i++) {
            auto r = result[i];
            r.push_back(n);
            result.push_back(r);
        }
        return result;
    }
};

File: 078/rust/main.rs
impl Solution {
    pub fn subsets(nums: Vec<i32>) -> Vec<Vec<i32>> {
        if nums.is_empty() {
            return vec![vec![]];
        }
        let mut nums = nums;
        let n = nums.pop().unwrap();
        let mut result = Solution::subsets(nums);
        let size = result.len();
        for i in 0..size {
            let mut r = result[i].clone();
            r.push(n);
            result.push(r);
        }
        return result;
    }
}
